storage/src/test/java/org/apache/kafka/storage/internals/log/RemoteIndexCacheTest.java

```diff
@@ -0,0 +1,1207 @@
+        Files.copy(entry.timeIndex().file().toPath(), Paths.get(Utils.replaceSuffix(entry.timeIndex().file().getPath(), "", tempSuffix)));
+
+        cache.remove(rlsMetadata.remoteLogSegmentId().id());
+
+        // wait until entry is marked for deletion
+        TestUtils.waitForCondition(entry::isMarkedForCleanup,
+                "Failed to mark cache entry for cleanup after invalidation");
+        TestUtils.waitForCondition(entry::isCleanStarted,
+                "Failed to cleanup cache entry after invalidation");
+
+        // restore index files
+        renameRemoteCacheIndexFileFromDisk(tempSuffix, remoteIndexCacheDir, tempSuffix);
+        // validate cache entry for the above key should be null
+        assertNull(cache.internalCache().getIfPresent(rlsMetadata.remoteLogSegmentId().id()));
+        cache.getIndexEntry(rlsMetadata);
+        // Index  Files already exist ,rsm should not fetch them again.
+        verifyFetchIndexInvocation(1);
+        // verify index files on disk
+        assertTrue(getIndexFileFromRemoteCacheDir(cache, LogFileUtils.INDEX_FILE_SUFFIX).isPresent(), "Offset index file should be present on disk at " + remoteIndexCacheDir.toPath());
+        assertTrue(getIndexFileFromRemoteCacheDir(cache, LogFileUtils.TXN_INDEX_FILE_SUFFIX).isPresent(), "Txn index file should be present on disk at " + remoteIndexCacheDir.toPath());
+        assertTrue(getIndexFileFromRemoteCacheDir(cache, LogFileUtils.TIME_INDEX_FILE_SUFFIX).isPresent(), "Time index file should be present on disk at " + remoteIndexCacheDir.toPath());
+    }
+
+    void renameRemoteCacheIndexFileFromDisk(String suffix, File remoteIndexCacheDir, String tempSuffix) throws IOException {
+        List<Path> paths = Files.walk(remoteIndexCacheDir.toPath())
+                .filter(Files::isRegularFile)
+                .filter(path -> path.getFileName().toString().endsWith(suffix)).toList();
+        for (Path path : paths) {
+            Utils.atomicMoveWithFallback(path, path.resolveSibling(path.getFileName().toString().replace(tempSuffix, "")));
+        }
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = IndexType.class, names = {"OFFSET", "TIMESTAMP", "TRANSACTION"})
+    public void testRSMReturnCorruptedIndexFile(IndexType testIndexType) throws RemoteStorageException {
+        when(rsm.fetchIndex(any(RemoteLogSegmentMetadata.class), any(IndexType.class))).thenAnswer(ans -> {
+            RemoteLogSegmentMetadata metadata = ans.getArgument(0);
+            IndexType indexType = ans.getArgument(1);
+            OffsetIndex offsetIdx = createOffsetIndexForSegmentMetadata(metadata, tpDir);
+            TimeIndex timeIdx = createTimeIndexForSegmentMetadata(metadata, tpDir);
+            TransactionIndex txnIdx = createTxIndexForSegmentMetadata(metadata, tpDir);
+            maybeAppendIndexEntries(offsetIdx, timeIdx);
+            // Create corrupt index file return from RSM
+            createCorruptedIndexFile(testIndexType, tpDir);
```

Comment:
nit: using `indexType` can honor the passed type.

Reply:
If I pass `indexType` then it's always `OFFSET`. I think this needs to be `testIndexType`, as it was in Scala, to trigger all the code paths. Granted this test is a bit convoluted and could probably be refactored.